sympy/combinatorics/fp_groups.py

```diff
@@ -1081,12 +1200,22 @@ def define_schreier_generators(C):
             C.P[beta][C.A_dict_inv[x]] = "<identity>"
             gamma += 1
         elif x in X and C.P[alpha][C.A_dict[x]] is None:
+            if rel:
+                # The schreier generators are defined as
+                # coset_rep(alpha)*x = schrier_gen*coset_rep(beta)
+                rh = C.coset_representative(alpha)*x
+                lh_suffix = C.coset_representative(beta)
+                for i in range(1, len(rh)):
+                    if rh.subword(i, len(rh)) == lh_suffix:
+                        _gens.append(rh.subword(0, i))
```

Comment:
You could just define the transversal as you go in some dictionary `tau`, much like what's done in the description of `DefineScreierGenerators` in the Handbook. Initially you set `tau[1] = f.identity`, then `tau[beta] = tau[alpha]*x` when `beta=gamma` and at this stage you can just do `_gens.append(tau(alpha)*x*tau(beta)**-1)`

Reply:
Yeah, I've changed it. I just haven't pushed that yet. Also, can we reduce the word after `tau(alpha)*x*tau(beta)**-1` is computed?

Reply:
Yes, of course, it'd be the same word as far as our group is concerned. Just check that the reduction time doesn't outweigh the benefits of having a shorter image.